app/services/hyrax/workflow/sipity_actions_generator.rb

```diff
@@ -38,11 +39,39 @@ def call
       private
 
         def generate_state_diagram!
+          # look for old states that need to be removed and validate the states can be removed.
+          removed_states = validate_states_to_be_removed
```

Comment:
@mjgiarlo If we compare the new states to the old states after we update the workflow then we will never see the changes.  The old state will  be the new state. at line 57.  The other issue is that we can not delete the states until they are no longer referenced by actions.